src/ansys/fluent/core/services/field_data.py

```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class _FieldDataBase:
+    def __init__(self, i_d, data):
+        self._data = data
+        self._id = i_d
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def surface_id(self):
+        return self._id
+
+    @property
+    def size(self):
+        return len(self._data)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+
+class ScalarFieldData(_FieldDataBase):
+    class _ScalarData:
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        super().__init__(i_d, [ScalarFieldData._ScalarData(_data) for _data in data])
+
+
+class Vector:
+    def __init__(self, x, y, z):
+        self._x = x
+        self._y = y
+        self._z = z
+
+    @property
+    def x(self):
+        return self._x
+
+    @property
+    def y(self):
+        return self._y
+
+    @property
+    def z(self):
+        return self._z
+
+
+class VectorFieldData(_FieldDataBase):
+    class _VectorData(Vector):
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data, scale):
+        data.shape = data.size // 3, 3
```

Comment:
@prmukherj should we raise an exception `if data.size % 3` ?

Reply:
Perhaps also some type hints would help readability?

Reply:
Done, Thank you.